beacon_node/store/src/reconstruct.rs

```diff
@@ -145,10 +144,8 @@ where
                             });
                         }
 
-                        self.compare_and_set_anchor_info_with_write(
-                            old_anchor,
-                            ANCHOR_FOR_ARCHIVE_NODE,
```

Comment:
Setting this const here innadvertedly mutated the anchor_slot, which would have broken HDiff reads

Reply:
Not doing this for now, seeing as anchor_slot seems to be working :crossed_fingers: